Spring-Boot/Spring-Boot-Problem/src/test/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemControllerTest.java

```diff
@@ -770,20 +817,36 @@ private Problem givenProblemWithTags(int problemId, String... tags) {
     private void filterProblemsWithTagsShouldContain(Token token, List<String> tags, List<ProblemItem> problemItems) throws Exception {
         String tagsSplitByCommas = String.join(", ", tags);
         mockMvc.perform(get(API_PREFIX)
-                .header("Authorization", bearerWithToken(token.getToken()))
-                .queryParam("tags", tagsSplitByCommas))
+                        .header("Authorization", bearerWithToken(token.getToken()))
+                        .queryParam("tags", tagsSplitByCommas))
                 .andExpect(status().isOk())
                 .andExpect(content().contentType(MediaType.APPLICATION_JSON))
                 .andExpect(content().json(objectMapper.writeValueAsString(problemItems)));
     }
 
-
     private List<ProblemView> getProblemsByTagsAndPage(Token token, int page, String... tags) throws Exception {
         String tagsSplitByCommas = String.join(",", tags);
         return getBody(mockMvc.perform(get(API_PREFIX)
-                .header("Authorization", bearerWithToken(token.getToken()))
-                .queryParam("tags", tagsSplitByCommas)
-                .queryParam("page", String.valueOf(page)))
+                        .header("Authorization", bearerWithToken(token.getToken()))
+                        .queryParam("tags", tagsSplitByCommas)
+                        .queryParam("page", String.valueOf(page)))
+                .andExpect(status().isOk()), new TypeReference<>() {
+        });
+    }
+
+    private List<ProblemItem> getInvisibleProblems(Token token) throws Exception {
+        return getBody(mockMvc.perform(get(API_PREFIX)
+                        .header("Authorization", bearerWithToken(token.getToken()))
+                        .queryParam("visible", String.valueOf(false))
+                        .queryParam("archive", String.valueOf(false)))
```

Comment:
Since the `archive` is not a required parameter, we can remove this line.

Reply:
Invisible problems do not include ```archive``` problems